src/state/src/State.c

```diff
@@ -158,6 +158,12 @@ STATUS stepStateMachine(PStateMachine pStateMachine)
         pStateMachineImpl->context.retryCount = 0;
         pStateMachineImpl->context.time = time;
 
+        // Call the state machine error handler
+        // This call could be a no-op if the state transition is happening for SERVICE_CALL_RESULT_OK code
+        if (pStateMachineImpl->context.pCurrentState->errorHandlerStateFunc != NULL) {
+            pStateMachineImpl->context.pCurrentState->errorHandlerStateFunc(pStateMachineImpl->customData);
```

Comment:
The state transition function returns a status. You might want to return the status from here if it's not STATUS_SUCCESS - have a way to control the state machine execution from the "hook"

Reply:
Yeah sure. Wrapped it in CHK_STATUS macro.